scripted-diff: Rename key size consts to be relative to their class

Empact opened the pull request:
~~And introduce CPubKeySig to host code relative to key sigs.~~

jonasschnelli commented on src/pubkey.h:
```diff
@@ -26,40 +26,48 @@ class CKeyID : public uint160
 
 typedef uint256 ChainCode;
 
+class CPubKeySig
```
Why not a namespace?

jonasschnelli commented:
Concept ACK.
Not sure if `CPubKeySig` needs its own space.

Empact commented:
Thanks, made PubKeySig a namespace.

Empact commented:
On second thought, dropped PubKeySig - without any other code it's barely a separate concern.

instagibbs commented:
seemingly unrelated travis error:
>File "/home/travis/build/bitcoin/bitcoin/build/bitcoin-i686-pc-linux-gnu/test/functional/rpc_net.py", line 54, in _test_getnettotals
>    assert_equal(before['bytesrecv_per_msg']['pong'] + 32, after['bytesrecv_per_msg']['pong'])

kicked travis

Empact commented:
@instagibbs thanks - seems like that is a timing-related failure, I pushed #12545 to increase the wait, and reduce the likelihood of spurious failures.

MarcoFalke commented:
Needs rebase if still relevant

Empact commented:
Rebased

Empact committed:
scripted-diff: Rename CPubKey and CKey::*_KEY_SIZE and COMPRESSED_*_KEY_SIZE

To SIZE and COMPRESSED_SIZE

-BEGIN VERIFY SCRIPT-
sed -i 's/PRIVATE_KEY_SIZE/SIZE/g' src/*.h src/*.cpp src/**/*.h src/**/*.cpp
sed -i 's/COMPRESSED_PRIVATE_KEY_SIZE/COMPRESSED_SIZE/g' src/*.h src/**/*.cpp src/**/*.h src/**/*.cpp
sed -i 's/PUBLIC_KEY_SIZE/SIZE/g' src/*.h src/*.cpp src/**/*.h src/**/*.cpp
sed -i 's/COMPRESSED_PUBLIC_KEY_SIZE/COMPRESSED_SIZE/g' src/*.h src/*.cpp src/**/*.h src/**/*.cpp
-END VERIFY SCRIPT-

Empact commented:
Rebased & reimplemented as a scripted-diff

sipa commented:
utACK f1583f26f2d9cb199ace6054a67797e1d4d9bed0

Empact commented:
Rebased

Empact commented:
Rebased to confirm scripted-diff still applies. It does. 👍

MarcoFalke commented:
utACK 2dda5dd034, but please note that the scripted diff will modify files outside the git tree. I'd prefer to avoid that.

Empact commented:
Updated to include 2 new cases, and brought it down to a single `sed` call which only acts on files in the repo.

Empact commented:
Rebased for #14195

ken2812221 commented on src/pubkey.h:
```diff
@@ -33,33 +33,33 @@ class CPubKey
-    static constexpr unsigned int PUBLIC_KEY_SIZE             = 65;
-    static constexpr unsigned int COMPRESSED_PUBLIC_KEY_SIZE  = 33;
+    static constexpr unsigned int SIZE             = 65;
+    static constexpr unsigned int COMPRESSED_SIZE  = 33;
```
nit: Could have another commit to fix the format here

ken2812221 reviewed (approved):
utACK 40c69642bd4bed49d17b75e3f544afbc8d10e8ae

Empact committed:
Fixup whitespace

Empact commented:
Rebased, added a commit to fixup whitespace.

meshcollider commented:
utACK https://github.com/bitcoin/bitcoin/pull/12461/commits/0580f86bb48004b797d4cb6273e7ffee0b0a0584
